org.eclipse.m2e.core.tests/src/org/eclipse/m2e/core/IBiConsumer.java

```diff
@@ -0,0 +1,44 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Georg Tsakumagos and others
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *      Georg Tsakumagos - initial declaration
+ *******************************************************************************/
+package org.eclipse.m2e.core;
+
+import java.util.function.Consumer;
+
+import org.eclipse.core.runtime.CoreException;
+
+/**
+ * Represents an operation that accepts two input arguments, returns no
+ * result and throws a {@link CoreException}. 
+ * This is the two-arity specialization of {@link Consumer}.
+ * Unlike most other functional interfaces, {@code BiConsumer} is expected
+ * to operate via side-effects.
+ *
+ *
+ * @param <T> the type of the first argument to the operation
+ * @param <U> the type of the second argument to the operation
+ * 
+ * @see Consumer
+ * @author Georg Tsakumagos
+ */
+@FunctionalInterface
+public interface IBiConsumer<T, U> {
```

Comment:
```suggestion
public interface CoreBiConsumer<T, U> {
```

Reply:
Okay sounds better. Same applies to the supplier.